lib/chef/application.rb

```diff
@@ -205,7 +205,7 @@ def configure_log_location
     # Based on config and whether or not STDOUT is a tty, should we setup a
     # secondary logger for stdout?
     def want_additional_logger?
-      Chef::Config.is_default?(:log_location) && Chef::Config[:log_location].tty? && !Chef::Config[:daemonize]
+      !(Chef::Config.is_default?(Chef::Config[:log_location]) || STDOUT.closed? || Chef::Config[:daemonize])
```

Comment:
changing the boolean logic here is also something that i'm pretty sure i investigated already and doesn't work correctly in the other use case, pretty certain that means you've changed the meaning of this check.

`Chef::Config[:log_location].class == IO && Chef::Config[:log_location].tty? && !Chef::Config[:daemonize]` might work correctly in both use cases and not break anything important.

If we have both an actual IO which is a tty then its very likely the log_location is STDOUT.  Any filename (String) or logger instance here would break this, along with even a File object that inherits from IO (not even sure if we support this, but I want to reduce the blast radius of this change to as small as possible).

Reply:
Thanks @lamont-granquist for review and suggestions, It gave me an idea to not override the `is_default?` instead trying out to change the code as per your suggestion.

Although the code `Chef::Config[:log_location].class == IO && Chef::Config[:log_location].tty? && !Chef::Config[:daemonize]` didn't seem to work out and we still get the duplicate loggers for `STDOUT`.

But if we change it to `!(Chef::Config[:log_location].class == IO || STDOUT.closed? || Chef::Config[:daemonize])` then this worked out well for `STDOUT` and `path/to/file`. 

Tested it with `chef-client`, `chef-solo` and `chef-apply`.
Please let me know your thoughts about this and we can then push the changes.

Reply:
Yes @lamont-granquist, I've tried with this code change as well and it is working fine.

`( Chef::Config[:log_location].class != IO ) && STDOUT.tty? && !Chef::Config[:daemonize]` works out well with `chef-client`, `chef-apply` & `chef-solo` for `STDOUT` and `/path/to/file`

Reply:
that is the one to go with then.  it should preserve the old code behavior.  the way it used to look in v12.0.0 was:

```
( Chef::Config[:log_location] != STDOUT ) && STDOUT.tty? && (!Chef::Config[:daemonize]) && (Chef::Config[:force_logger])
```

the force_logger bit is no longer relevant.  we've now preserved the first three conditional -- only the first conditional has been changed to != on the class being IO rather than != on the instance being STDOUT.

i think i'm good with that.